### PR TITLE
Adopted pets: Show Incomplete Tasks

### PR DESCRIPTION
**Pull request by sarvaiyanidhi**
# 🔗 Issue
#961

# ✍️ Description

- Change existing functionality of displaying files in adopted_pets from turbo_stream to turbo_frame
- Add additional functionality to show incomplete tasks for pet along with files details

# 📷 Screenshots/Demos

alta_adopted_pet_incomplete_tasks

### CODE REVIEW

**Comment by sarvaiyanidhi** [comment]
Hi @kasugaijin ,

As we discussed earlier in issue, I am loading tasks as well using turbo_frame src attribute when adopter clicks on Pet card to load files. Please check the approach and let me know if you think we need to change anything. Thanks

File: app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb [hunk 1]
```diff
@@ -7,14 +7,8 @@ class FilesController < Organizations::BaseController
...
-            format.html # Regular HTML response
-            format.turbo_stream do
-              render turbo_stream: turbo_stream.replace(
-                "pet_files",
-                partial: "organizations/shared/file_attachment_table",
-                locals: {pet: @pet}
-              )
-            end
+            format.html
+            format.turbo_stream
```
Comment (kasugaijin) [inline app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb:11]: Do we need to define formats for turbo frame?

File: app/views/organizations/adopter_fosterer/adopted_pets/tasks/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<turbo-frame id="pet_tasks">
+ <ul class="list-group">
+<%= render "organizations/shared/incomplete_tasks_table", locals: {task: @task} %>
+</ul>
+  
+</turbo-frame>
```
Comment (kasugaijin) [inline app/views/organizations/adopter_fosterer/adopted_pets/tasks/index.html.erb:6]: New line :)

File: app/views/organizations/adopter_fosterer/adopted_pets/index.html.erb [hunk 3]
```diff
@@ -21,8 +21,8 @@
         <% end %>
       </div>
     </div>
-    <div id="pet_files">
+    <turbo-frame id="pet_files">
       <!-- This will be replaced with the pet files table when a pet is clicked -->
-    </div>
+    </turbo-frame>
   <% end %>
 <% end %>
```
Comment (kasugaijin) [inline app/views/organizations/adopter_fosterer/adopted_pets/index.html.erb:28]: new line :)

File: app/views/organizations/shared/_incomplete_tasks_table.html.erb
Comment (kasugaijin) [inline app/views/organizations/shared/_incomplete_tasks_table.html.erb:1]: I think this partial can live in the adopter fosterer scope?

File: app/views/organizations/adopter_fosterer/adopted_pets/files/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  <ul class="nav nav-pills mb-3" id="pills-tab" role="tablist">
+    <li class="nav-item" role="presentation">
+      <button class="nav-link active" id="pills-files-tab" data-bs-toggle="pill" data-bs-target="#pills-files" type="button" role="tab" aria-controls="pills-files" aria-selected="true">Files</button>
+    </li>
+    <li class="nav-item" role="presentation">
+      <button class="nav-link" id="pills-tasks-tab" data-bs-toggle="pill" data-bs-target="#pills-tasks" type="button" role="tab" aria-controls="pills-tasks" aria-selected="false">Tasks</button>
+    </li>
+    
+  </ul>
+  <div class="tab-content" id="pills-tabContent">
```
Comment (kasugaijin) [inline app/views/organizations/adopter_fosterer/adopted_pets/files/index.html.erb:11]: Ah I really like the use of tabs here! Good choice

File: app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    module AdoptedPets
+      class TasksController < Organizations::BaseController
+        before_action :context_authorize!
+
+        def index
+          @pet = Pet.find(params[:adopted_pet_id])
+          @tasks = @pet.tasks.where(completed: false)
+
+          respond_to do |format|
+            format.html
```
Comment (kasugaijin) [inline app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb:12]: same here re formats

**Review by kasugaijin — COMMENTED** [review]
This is looking really good @sarvaiyanidhi - just a few comments

File: app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb (hunk 1 above)
Comment (sarvaiyanidhi) [reply app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb:11]: We don't need it. Will remove it

File: app/views/organizations/adopter_fosterer/adopted_pets/files/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="tab-pane fade show active" id="pills-files" role="tabpanel" aria-labelledby="pills-files-tab">
+      <%= render "organizations/shared/file_attachment_table",
+                  locals: {pet: @pet} %>
+    </div>
+    <div class="tab-pane fade" id="pills-tasks" role="tabpanel" aria-labelledby="pills-tasks-tab">
+      <%= turbo_frame_tag "pet_tasks", src: adopter_fosterer_adopted_pet_tasks_path(@pet) %>
```
[the remark this answers is not in the document]
Comment (sarvaiyanidhi) [reply app/views/organizations/adopter_fosterer/adopted_pets/files/index.html.erb]: Lazy loading works.. Have updated code. Thanks for feedback

File: app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb [hunk 7]
```diff
@@ -7,14 +7,7 @@ class FilesController < Organizations::BaseController
...
           respond_to do |format|
-            format.html # Regular HTML response
-            format.turbo_stream do
-              render turbo_stream: turbo_stream.replace(
-                "pet_files",
-                partial: "organizations/shared/file_attachment_table",
-                locals: {pet: @pet}
-              )
-            end
+            format.html
```
Comment (kasugaijin) [inline app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb:10]: I think we can remove this as it’s the default format?

File: app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  module AdopterFosterer
+    module AdoptedPets
+      class TasksController < Organizations::BaseController
+        before_action :context_authorize!
+
+        def index
+          @pet = Pet.find(params[:adopted_pet_id])
+          @tasks = @pet.tasks.where(completed: false)
+
+          respond_to do |format|
```
Comment (kasugaijin) [inline app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb:11]: Same here I think we can remove this whole block

File: app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+module Organizations
+  module AdopterFosterer
+    module AdoptedPets
+      class TasksController < Organizations::BaseController
+        before_action :context_authorize!
+
+        def index
+          @pet = Pet.find(params[:adopted_pet_id])
+          @tasks = @pet.tasks.where(completed: false)
```
Comment (kasugaijin) [inline app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb:9]: We can use this scope I think
https://github.com/rubyforgood/pet-rescue/blob/007914e4be525f35e245cee9bb909f3af052ddb2/app/models/task.rb#L35

File: test/controllers/organizations/adopter_fosterer/adopted_pets/files_controller_test.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+require "test_helper"
+require "action_policy/test_helper"
+
+class Organizations::AdopterFosterer::AdoptedPets::FilesControllerTest < ActionDispatch::IntegrationTest
```
Comment (kasugaijin) [inline test/controllers/organizations/adopter_fosterer/adopted_pets/files_controller_test.rb:4]: Can we nest these modules?

File: test/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller_test.rb [hunk 11]
```diff
@@ -0,0 +1,36 @@
+require "test_helper"
+require "action_policy/test_helper"
+
+class Organizations::AdopterFosterer::AdoptedPets::TasksControllerTest < ActionDispatch::IntegrationTest
```
Comment (kasugaijin) [inline test/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller_test.rb:4]: Same here for nesting modules

**Review by kasugaijin — COMMENTED** [review]
Looking great - just a couple of small comments

File: test/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller_test.rb (hunk 11 above)
Comment (sarvaiyanidhi) [reply test/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller_test.rb]: Done

File: test/controllers/organizations/adopter_fosterer/adopted_pets/files_controller_test.rb (hunk 10 above)
Comment (sarvaiyanidhi) [reply test/controllers/organizations/adopter_fosterer/adopted_pets/files_controller_test.rb]: Done

File: app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb (hunk 9 above)
Comment (sarvaiyanidhi) [reply app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb]: Yeah, I missed this scope.. Have added it now.

File: app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb (hunk 8 above)
Comment (sarvaiyanidhi) [reply app/controllers/organizations/adopter_fosterer/adopted_pets/tasks_controller.rb]: Done

File: app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb (hunk 7 above)
Comment (sarvaiyanidhi) [reply app/controllers/organizations/adopter_fosterer/adopted_pets/files_controller.rb]: Done

**Review by kasugaijin — APPROVED** [review]
Thank you! This is great. We'll do exactly the same for the fosterer user now, too, and this will be a very useful implementation to follow.

**Pull request merged**